execute/tokendata/usdc/usdc.go

```diff
@@ -4,43 +4,168 @@ import (
 	"context"
 	"fmt"
 
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"golang.org/x/exp/maps"
+
 	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
 	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
 )
 
 type USDCCCTPTokenDataObserver struct {
```

Comment:
This name is a mouthful, two questions:

* any reason this needs to be a public struct?
* can this just be `TokenDataObserver` since we're in the USDC package? The "USDCCCTP" prefix seem redundant

similar comments apply elsewhere where this phenomenon is present. If it can be private then just `tokenDataObserver` is good enough IMO. Its perfectly fine to return private structs, see https://go.dev/doc/effective_go#generality

Reply:
yeah, it can be private as long as we expose actors, will do

Reply:
> similar comments apply elsewhere where this phenomenon is present. If it can be private then just tokenDataObserver is good enough IMO. Its perfectly fine to return private structs, see https://go.dev/doc/effective_go#generality

Actually, our linter settings don't allow me to return private structs from constructors

```go
execute/tokendata/usdc/usdc.go:26:3: unexported-return: exported func NewTokenDataObserver returns unexported type *usdc.tokenDataObserver, which can be annoying to use (revive)
) *tokenDataObserver {
```